var/spack/repos/builtin/packages/chameleon/package.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Chameleon(CMakePackage, CudaPackage):
+    """Dense Linear Algebra for Scalable Multi-core Architectures and GPGPUs"""
+    homepage = "https://gitlab.inria.fr/solverstack/chameleon"
+    url      = "https://gitlab.inria.fr/solverstack/chameleon/uploads/b299d6037d7636c6be16108c89bc2aab/chameleon-1.1.0.tar.gz"
+    git      = "https://gitlab.inria.fr/solverstack/chameleon.git"
+
+    version('master', branch='master', submodules=True)
+    version('1.1.0', 'e64d0438dfaf5effb3740e53f3ab017d12744b85a138b2ef702a81df559126df')
+
+    # cmake's specific
+    variant('shared', default=True, description='Build chameleon as a shared library')
+
+    # chameleon's specific
+    variant(
+        'runtime', default='starpu', description='Runtime support',
+        values=('openmp', 'starpu'), multi=False
+    )
+    variant('mpi', default=True, description='Enable MPI')
+    variant('cuda', default=False, description='Enable CUDA')
+    variant('fxt', default=False, description='Enable FxT tracing support through StarPU')
+    variant('simgrid', default=False, description='Enable simulation mode through StarPU+SimGrid')
+
+    # dependencies
+    depends_on("pkg-config", type='build')
+    depends_on("blas", when='~simgrid')
+    depends_on("lapack", when='~simgrid')
+    depends_on("starpu", when='runtime=starpu')
+    depends_on("starpu~mpi", when='~mpi runtime=starpu')
+    depends_on("starpu+cuda", when='+cuda ~simgrid runtime=starpu')
+    depends_on("starpu+fxt", when='+fxt runtime=starpu')
+    depends_on("starpu+simgrid", when='+simgrid runtime=starpu')
+    depends_on("starpu+mpi~shared+simgrid", when='+mpi +simgrid runtime=starpu')
+    depends_on("mpi", when='+mpi ~simgrid')
+    depends_on("cuda", when='+cuda ~simgrid')
+    depends_on("fxt", when='+fxt runtime=starpu')
+
+    def cmake_args(self):
+
+        spec = self.spec
+        args = []
+
+        args.extend([
+            "-Wno-dev",
+            "-DCMAKE_COLOR_MAKEFILE:BOOL=ON",
+            "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON",
+            "-DCHAMELEON_ENABLE_EXAMPLE=ON",
+            "-DCHAMELEON_ENABLE_TESTING=ON",
+            "-DBUILD_SHARED_LIBS=%s " % ('ON' if '+shared' in spec else 'OFF'),
+            "-DCHAMELEON_USE_MPI=%s " % ('ON' if '+mpi' in spec else 'OFF'),
+            "-DCHAMELEON_USE_CUDA=%s " % ('ON' if '+cuda' in spec else 'OFF'),
+            "-DCHAMELEON_SIMULATION=%s " % ('ON' if '+simgrid' in spec else 'OFF')
+        ])
```

Comment:
Beware of typos, since I didn't try the code, but this can be written using helper functions:
```suggestion
        args = [
            "-Wno-dev",
            self.define("CMAKE_COLOR_MAKEFILE", "ON"),
            self.define("CMAKE_VERBOSE_MAKEFILE", ON"),
            self.define("CHAMELEON_ENABLE_EXAMPLE", "ON"),
            self.define("CHAMELEON_ENABLE_TESTING", "ON"),
            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
            self.define_from_variant("CHAMELEON_USE_MPI", "mpi"),
            self.define_from_variant("CHAMELEON_USE_CUDA",  "cuda"),
            self.define_from_variant("CHAMELEON_SIMULATION", "simgrid")
        ]
```
Similarly below.